
<launch>
    <arg name="icp_config"/>
    <arg name="working_directory" default="$(env PWD)" />

    <include file="/home/ubuntu/catkin_ws/src/zed-ros-wrapper-master/launch/zed.launch">
        <param name="frequency" value="10" />
    </include>

    <node name="cloud_matcher" pkg="pointmatcher_ros" type="matcher_service" >
      <param name="config" type="str" value="$(arg icp_config)" />
    </node>

    <node name="zedtobase_repeat" pkg="tf" type="static_transform_publisher" args="0 0 0.5 0 0 0 base_footprint zed_current_frame 100" />

    <node name="command_repeater" pkg="husky_trainer" type="command_repeater">
        <param name="input" value="/teach_repeat/desired_command" />
        <param name="output" value="/mobile_base/commands/velocity" />
    </node>
    <node name="repeat_node" pkg="husky_trainer" type="repeat" output="screen">
        <param name="working_directory" value="$(arg working_directory)" />
        <param name="readings_topic" value="/zed/point_cloud/cloud_registered" />
        <param name="_lambda_x" value="1.0" />
    </node>
</launch>
